File: C/703A-Mishka and Game.c
/*
    Solution to Codeforces problem 703A
    Copyright (c) GeneralMing. All rights reserved.

    https://github.com/GeneralMing/codeforces
*/

#include<stdio.h>
#include<string.h>

int main()
{
	int n, m = 0, c = 0, die1, die2;
	scanf("%d", &n);
	for(int i = 0; i < n; i++)
	{
		scanf("%d%d", &die1, &die2);
		if(die1 > die2)
			m++;
		else if(die2 > die1)
			c++;
	}
	if(m > c)
		printf("Mishka\n");
	else if(c > m)
		printf("Chris\n");
	else
		printf("Friendship is magic!^^\n");

}

File: Python/703A-Mishka and Game.py
"""
    Solution to Codeforces problem 703A
    Copyright (c) GeneralMing. All rights reserved.

    https://github.com/GeneralMing/codeforces
"""

n = int(input())
m = 0
c = 0
for i in range(n):
	die1, die2 = input().split(" ")
	die1 = int(die1)
	die2 = int(die2)
	if(die1 > die2):
		m += 1
	elif(die2 > die1):
		c += 1
if(m > c):
	print("Mishka")
elif(c > m):
	print("Chris")
else:
	print("Friendship is magic!^^")